asynciterator.js

```diff
@@ -529,6 +604,13 @@ ArrayIterator.prototype._toStringDetails = function () {
   return '(' + (this._buffer && this._buffer.length || 0) + ')';
 };
 
+/* Called by {@link AsyncIterator#destroy} */
+ArrayIterator.prototype._destroy = function (error, callback) {
+  // Release our complete buffer.
+  this._buffer = null;
```

Comment:
Or empty array? (Might be faster since then there's only one internal V8 class). However, I do see `this._buffer` checks above, so might be null in other cases.

Reply:
There's a `delete this._buffer` call above, which is better indeed. I'll do the same here.